service/pki/service.go

```diff
@@ -118,73 +118,6 @@ func (s *service) IsMounted(clusterID string) (bool, error) {
 	return true, nil
 }
 
-func (s *service) IsRoleCreated(clusterID string) (bool, error) {
-	// Create a client for the logical backend configured with the Vault token
-	// used for the current cluster's PKI backend.
-	logicalBackend := s.VaultClient.Logical()
-
-	// Check if a PKI for the given cluster ID exists.
-	secret, err := logicalBackend.List(s.ListRolesPath(clusterID))
-	if IsNoVaultHandlerDefined(err) {
-		return false, nil
-	} else if err != nil {
-		return false, microerror.Mask(err)
-	}
-
-	// In case there is not a single role for this PKI backend, secret is nil.
-	if secret == nil {
-		return false, nil
-	}
-
-	// When listing roles a list of role names is returned. Here we iterate over
-	// this list and if we find the desired role name, it means the role has
-	// already been created.
-	if keys, ok := secret.Data["keys"]; ok {
-		if list, ok := keys.([]interface{}); ok {
-			for _, k := range list {
-				if str, ok := k.(string); ok && str == s.RoleName(clusterID) {
-					return true, nil
-				}
-			}
-		}
-	}
-
-	return false, nil
-}
-
-func (s *service) VerifyPKISetup(clusterID string) (bool, error) {
```

Comment:
The difference in creating and removing code is that one is constructive and the other disruptive. When you remove code you break the API and as we found out some service depends on this. So this change should not just come with something else but should rather be sorted out separately. When you want to remove code here and you want to fix all the places where this code was used then go for it. I just think that it would be good to focus on one thing and sort out the rest later. Up to you.